google-ads-examples/src/main/java/com/google/ads/googleads/examples/targeting/AddDemographicTargetingCriteria.java

```diff
@@ -0,0 +1,138 @@
+
+/** Adds demographic criteria to an ad group. */
+public class AddDemographicTargetingCriteria {
+  private static class AddDemographicTargetingCriteriaParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private Long customerId;
+
+    @Parameter(names = ArgumentNames.AD_GROUP_ID, required = true)
+    private Long adGroupId;
+  }
+
+  public static void main(String[] args) {
+    AddDemographicTargetingCriteriaParams params = new AddDemographicTargetingCriteriaParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.adGroupId = Long.parseLong("INSERT_AD_GROUP_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new AddDemographicTargetingCriteria()
+          .runExample(
+              googleAdsClient,
+              params.customerId,
+              params.adGroupId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID in which to create criterion.
+   * @param adGroupId the ad group ID in which to create criterion.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId, long adGroupId) {
+    String adGroupResourceName = ResourceNames.adGroup(customerId, adGroupId);
+
+    List<AdGroupCriterionOperation> operations = new ArrayList<>();
+
+    // Creates a gender demographic criteria.
+    AdGroupCriterion genderAdGroupCriterionOperation = AdGroupCriterion.newBuilder()
+        .setAdGroup(StringValue.of(adGroupResourceName))
+        .setGender(GenderInfo.newBuilder().setType(GenderType.MALE).build())
```

Comment:
@fiboknacky I see this is based on the PHP example, which in turn was based on the AdWords API example. However, since demo criteria such as genders are a discrete set, adding a demo criteria without also setting a bid modifier or a bid doesn't serve much of a purpose. Setting a bid may be confusing because that brings in the whole question of [TargetRestriction](https://developers.google.com/google-ads/api/reference/rpc/google.ads.googleads.v2.common#google.ads.googleads.v2.common.TargetRestriction), but WDYT of having this at least apply a bid _modifier_?

Reply:
Waiting on @fiboknacky response, but I've made that change for the time being.

Reply:
SGTM.

Reply:
Very sorry for the back and forth here, but I got curious and tried out this code example, and what I found was that adding a positive `Gender` criterion at the ad group level implicitly excludes all other `Gender` criteria that were not explicitly added at the ad group level. The same applies to `AgeRange` and `ParentalStatus`.

In other words, the behavior is:
*  If you don't explicitly target or exclude any `Gender` criteria, all `Gender` criteria are implicitly targeted.
*  If you target any `Gender` criteria by adding an `AdGroupCriterion`, all other criteria you did not explicitly add are implicitly _excluded_.

In light of this, my suggestion to add a bid modifier doesn't necessarily make sense, as we're only targeting one `Gender`. Could you revert the addition I suggested?

Thanks

Reply:
Done.